src/java.base/share/classes/java/lang/String.java

```diff
@@ -1284,29 +1308,43 @@ private static int malformed4(byte[] src, int sp) {
         return 3;
     }
 
-    private static void throwMalformed(int off, int nb) {
+    @SuppressWarnings("unchecked")
+    private static <E extends Exception> E malformedInputException(int off, int nb, Class<E> exceptionType) {
+        MalformedInputException mie = new MalformedInputException(nb);
         String msg = "malformed input off : " + off + ", length : " + nb;
-        throw new IllegalArgumentException(msg, new MalformedInputException(nb));
+        mie.initCause(new IllegalArgumentException(msg));
+        return (E) mie;
     }
 
-    private static void throwMalformed(byte[] val) {
+    private static <E extends Exception> E malformedInputException(byte[] val, Class<E> exceptionType) {
```

Comment:
This one seems only meaningful for ascii. Should we rename this to maybe `malformedAscii`?